src/dftbp/dftbplus/initprogram.F90

```diff
@@ -4894,6 +4919,9 @@ subroutine getDenseDescBlacs(env, rowBlock, colBlock, denseDesc)
     call scalafx_getdescriptor(env%blacs%orbitalGrid, nn, nn, rowBlock, colBlock,&
         & denseDesc%blacsOrbSqr)
 
+    call scalafx_getdescriptor(env%blacs%rowOrbitalGrid, nn, nn, rowBlock * colBlock, 1,&
+        & denseDesc%blacsColumnSqr)
```

Comment:
Is `rowBlock * colBlock` equal to the number of columns? From my understanding, you will otherwise not get an entire row on the processor.
What happens anyway if the number of rows is different from the number of processors?

Reply:
Well spotted, yes this should be 
```suggestion
    call scalafx_getdescriptor(env%blacs%rowOrbitalGrid, nn, nn, nn, 1,&
        & denseDesc%blacsColumnSqr)
```
The [tester](https://github.com/dftbplus/scalapackfx/blob/main/test/test_gemr2d.f90) for scalapackfx was correct.

Reply:
If nRow /= nProc then some processors will have fewer rows stored than others (assuming n_matrix > nProc, which is always the case at the moment, excepting for the use of processor groups for k-points etc). The default of choosing a square BLACS processor grid doesn't seem to balance this well (probably NUMROC only really cares about cyclic block distribution, not the effective row cyclic here, at least for the ScaLAPACK implementations I've tested). Hence the psudo-1D processor grid.